test/unit/three_scale/analytics/user_tracking_test.rb

```diff
@@ -8,125 +8,176 @@ def setup
     @user = FactoryBot.build_stubbed(:user, account: account, email: 'foo@example.net')
   end
 
-  def test_track_uses_segment
-    UserTracking::Segment.expects(:track).once
-    UserTracking.new(@user).track('foo')
-  end
+  class SegmentEnabled < UserTrackingTest
+    def setup
+      super
+      Features::SegmentConfig.stubs(enabled?: true)
+      clean_segment_adapter
```

Comment:
Imo this line should be a teardown of `UserTrackingTest` 🤔 because:

1. We already expect it to be clean before starting, the thing is that we want to clean it at the end too so we don't unexpectedly affect other tests.
2. In `UserTrackingTest` because it affects both subclasses